### DISCUSSION TITLE
Suggestion to edit the readme: Start on boot with systemd

### DISCUSSION THREAD

**Discussion by Trix2Fix** [bluenviron/mediamtx#843] [General]
I suggest changing the following part in the readme description - "Start on boot with systemd" 

```
After=network-online.target
Wants=network-online.target
```

instead of 

```After=network.target```

Reason:
On my raspberry pi the server started before the network was up and closed because of that:

```
2022/02/17 02:08:39 INF rtsp-simple-server v0.0.0
2022/02/17 02:08:39 DEB path manager opened
2022/02/17 02:08:39 INF [RTSP] listener opened on :8554 (TCP)
2022/02/17 02:08:39 INF [HLS] listener opened on :8888
2022/02/17 02:08:39 ERR listen tcp: lookup stream.fritz.box on 192.168.1.1:53: dial udp 192.168.1.1:53: connect: network is unreachable
2022/02/17 02:08:39 INF [RTSP] listener is closing
2022/02/17 02:08:39 DEB path manager is closing
2022/02/17 02:08:39 INF [HLS] listener is closing
2022/02/17 02:08:39 INF waiting for external commands
```

```
pi@stream:~ $ systemctl status rtsp-simple-server
● rtsp-simple-server.service
     Loaded: loaded (/etc/systemd/system/rtsp-simple-server.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Thu 2022-02-17 02:08:39 CET; 9min ago
    Process: 474 ExecStart=/usr/local/bin/rtsp-simple-server /usr/local/etc/rtsp-simple-server.yml (code=exited, status=1/FAILURE)
   Main PID: 474 (code=exited, status=1/FAILURE)
        CPU: 154ms

Feb 17 02:08:38 stream systemd[1]: Started rtsp-simple-server.service.
Feb 17 02:08:39 stream systemd[1]: rtsp-simple-server.service: Main process exited, code=exited, status=1/FAILURE
Feb 17 02:08:39 stream systemd[1]: rtsp-simple-server.service: Failed with result 'exit-code'.
```

After changing it, the server starts fine. Here I got the tip:
[https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/](url)

**Comment by aler9** [comment]
The readme was updated, thanks.

**Comment by mdPlusPlus** [comment]
This seems to no longer be part of the Readme.md. Is this intentional?

Form what I could gather from here, I'd say it should say `Requires=network-online.target`: https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html

> Wants=
> 
>     Configures (weak) requirement dependencies on other units. This option may be specified more than once or multiple space-separated units may be specified in one option in which case dependencies for all listed names will be created. Dependencies of this type may also be configured outside of the unit configuration file by adding a symlink to a .wants/ directory accompanying the unit file. For details, see above.
> 
>     Units listed in this option will be started if the configuring unit is. However, if the listed units fail to start or cannot be added to the transaction, this has no impact on the validity of the transaction as a whole, and this unit will still be started. This is the recommended way to hook the start-up of one unit to the start-up of another unit.
> 
>     Note that requirement dependencies do not influence the order in which services are started or stopped. This has to be configured independently with the After= or Before= options. If unit foo.service pulls in unit bar.service as configured with Wants= and no ordering is configured with After= or Before=, then both units will be started simultaneously and without any delay between them if foo.service is activated.
> 
>     Added in version 201.
> Requires=
> 
>     Similar to Wants=, but declares a stronger requirement dependency. Dependencies of this type may also be configured by adding a symlink to a .requires/ directory accompanying the unit file.
> 
>     If this unit gets activated, the units listed will be activated as well. If one of the other units fails to activate, and an ordering dependency After= on the failing unit is set, this unit will not be started. Besides, with or without specifying After=, this unit will be stopped (or restarted) if one of the other units is explicitly stopped (or restarted).
> 
>     Often, it is a better choice to use Wants= instead of Requires= in order to achieve a system that is more robust when dealing with failing services.
> 
>     Note that this dependency type does not imply that the other unit always has to be in active state when this unit is running. Specifically: failing condition checks (such as ConditionPathExists=, ConditionPathIsSymbolicLink=, … — see below) do not cause the start job of a unit with a Requires= dependency on it to fail. Also, some unit types may deactivate on their own (for example, a service process may decide to exit cleanly, or a device may be unplugged by the user), which is not propagated to units having a Requires= dependency. Use the BindsTo= dependency type together with After= to ensure that a unit may never be in active state without a specific other unit also in active state (see below).
> 
>     Added in version 201.

**Reply by aler9** [reply]
the systemd part is still part of the README, it just has been moved to _Start on boot_, _Linux_:
https://github.com/bluenviron/mediamtx?tab=readme-ov-file#linux

[a comment by an automated account is not shown]
